### PR TITLE
feat: new check field in subscriptions for (not) submitting invoices

### PR DESCRIPTION
**Pull request by meike289**
Docs: https://github.com/frappe/erpnext_documentation/pull/293

Add new field "submit_invoice" in doctype subscriptions.

submit_invoice

If this field is checked, subscriptions will be submitted automatically (as before). But if it's not checked, subscriptions will not be submitted automatically anymore.

With some customers you have to change an invoice often afterwards. For example, if an additional item has to be added. Some customers need to have this item on the same invoice. It's very inconvenient to cancel the sales invoice.

Would it be also possible to backport this pull request for version 12?

### CODE REVIEW

**Comment by barredterra** [comment]
@mergifyio backport version-12-hotfix

**Comment by barredterra** [comment]
Hi @meike289 , PRs have to go against `develop` or `hotfix` branches. I changed this PR to go into `develop`. Mergify will attempt to make an automatic backport to `version-12-hotfix`. If it fails, you'll have to manually create a new PR for `version-12-hotfix`.

Please also update [the docs](https://github.com/frappe/erpnext_documentation/blob/master/erpnext_documentation/www/docs/user/manual/en/accounts/subscription.md) to reflect your changes and add a link to your docs PR in the description of this PR.

**Comment by meike289** [comment]
> Hi @meike289 , PRs have to go against `develop` or `hotfix` branches. I changed this PR to go into `develop`. Mergify will attempt to make an automatic backport to `version-12-hotfix`. If it fails, you'll have to manually create a new PR for `version-12-hotfix`.
> 
> Please also update [the docs](https://github.com/frappe/erpnext_documentation/blob/master/erpnext_documentation/www/docs/user/manual/en/accounts/subscription.md) to reflect your changes and add a link to your docs PR in the description of this PR.

Hey @barredterra, thank you for your comment and your help! I will update the docs immediately.

**Comment by barredterra** [comment]
Failing Semgrep is not related to this PR. We can squash this PR to get a semantic commit.

File: erpnext/accounts/doctype/subscription/subscription.py [hunk 1]
```diff
@@ -383,7 +383,9 @@ def create_invoice(self, prorate):
 
 		invoice.flags.ignore_mandatory = True
 		invoice.save()
-		invoice.submit()
+
+		if self.submit_invoice == 1:
```
Comment (ankush) [inline erpnext/accounts/doctype/subscription/subscription.py]:
```suggestion
		if self.submit_invoice:
```

**Review by barredterra — DISMISSED** [review]

**Review by barredterra — APPROVED** [review]

**Comment by barredterra** [comment]
@deepeshgarg007 can we get this merged?

**Review by deepeshgarg007 — APPROVED** [review]

**Comment by barredterra** [comment]
@meike289 it looks like the automatic backport failed, so you'll have to manually create a backport to the `version-12-hotifx` branch.

**Comment by meike289** [comment]
> @meike289 it looks like the automatic backport failed, so you'll have to manually create a backport to the `version-12-hotifx` branch.

Hey @barredterra and @deepeshgarg007, here is the backport: [#25560](https://github.com/frappe/erpnext/pull/25560)

**Pull request merged**